checker/src/org/checkerframework/checker/minlen/MinLenTransfer.java

```diff
@@ -0,0 +1,584 @@
+import org.checkerframework.dataflow.analysis.TransferResult;
+import org.checkerframework.dataflow.cfg.node.ArrayAccessNode;
+import org.checkerframework.dataflow.cfg.node.ArrayCreationNode;
+import org.checkerframework.dataflow.cfg.node.AssignmentNode;
+import org.checkerframework.dataflow.cfg.node.EqualToNode;
+import org.checkerframework.dataflow.cfg.node.FieldAccessNode;
+import org.checkerframework.dataflow.cfg.node.GreaterThanNode;
+import org.checkerframework.dataflow.cfg.node.GreaterThanOrEqualNode;
+import org.checkerframework.dataflow.cfg.node.LessThanNode;
+import org.checkerframework.dataflow.cfg.node.LessThanOrEqualNode;
+import org.checkerframework.dataflow.cfg.node.MethodInvocationNode;
+import org.checkerframework.dataflow.cfg.node.Node;
+import org.checkerframework.dataflow.cfg.node.NotEqualNode;
+import org.checkerframework.framework.flow.CFAbstractTransfer;
+import org.checkerframework.framework.flow.CFValue;
+import org.checkerframework.framework.type.AnnotatedTypeMirror;
+import org.checkerframework.framework.type.QualifierHierarchy;
+import org.checkerframework.javacutil.AnnotationUtils;
+import org.checkerframework.javacutil.TreeUtils;
+
+public class MinLenTransfer extends CFAbstractTransfer<CFValue, MinLenStore, MinLenTransfer> {
+
+    protected MinLenAnalysis analysis;
+    protected static MinLenAnnotatedTypeFactory atypeFactory;
+    protected final ExecutableElement listAdd;
+    protected final ExecutableElement listAdd2;
+    protected final ExecutableElement listToArray;
+    protected final ExecutableElement listToArray1;
+    protected final ExecutableElement arrayAsList;
+    protected static ValueAnnotatedTypeFactory valueAnnotatedTypeFactory;
+
+    private QualifierHierarchy qualifierHierarchy;
+
+    public MinLenTransfer(MinLenAnalysis analysis) {
+        super(analysis);
+        this.analysis = analysis;
+        atypeFactory = (MinLenAnnotatedTypeFactory) analysis.getTypeFactory();
+        valueAnnotatedTypeFactory = atypeFactory.getValueAnnotatedTypeFactory();
+        qualifierHierarchy = atypeFactory.getQualifierHierarchy();
+        ProcessingEnvironment env = atypeFactory.getProcessingEnv();
+        this.listAdd = TreeUtils.getMethod("java.util.List", "add", 1, env);
+        this.listAdd2 = TreeUtils.getMethod("java.util.List", "add", 2, env);
+        this.listToArray = TreeUtils.getMethod("java.util.List", "toArray", 0, env);
+        this.listToArray1 = TreeUtils.getMethod("java.util.List", "toArray", 1, env);
+        this.arrayAsList = TreeUtils.getMethod("java.util.Arrays", "asList", 1, env);
+    }
+
+    @Override
+    public TransferResult<CFValue, MinLenStore> visitAssignment(
+            AssignmentNode node, TransferInput<CFValue, MinLenStore> in) {
+        TransferResult<CFValue, MinLenStore> result = super.visitAssignment(node, in);
+
+        // When an array is created using another array's length as the dimension, transfer
```

Comment:
Despite that, when I changed the code, a test started failing: ArrayCopy.java in tests/minlen. I would expect this to be handled by the framework, but it's not; do you know why? The change I made was to keep the roughly the same code, but move it to visitArrayCreation instead of visitAssignment, and remove the first two if statements (which checked whether an array was being assigned into). I also made the obvious changes based on what node now is (i.e. node instead of (ArrayCreationNode)(node.getExpression()).